ts/output/common/Wrappers/scriptbase.ts

```diff
@@ -215,77 +264,119 @@ export function CommonScriptbaseMixin<
-              core.node.isKind('mstyle') || core.node.isKind('mpadded') ||
-              core.node.isKind('mphantom') || core.node.isKind('semantics'))) {
-        core = core.childNodes[0];
-        if (!core) return;
-      }
-      if (!('noIC' in core)) return;
-      this.baseCore = core;
       //
       //  Check if the base is a mi or mo that needs italic correction removed
       //
-      if (!(this.constructor as CommonScriptbaseClass).useIC) {
-        (core as CommonMo).noIC = true;
+      if (('noIC' in core) && !(this.constructor as CommonScriptbaseClass).useIC) {
+        (core as unknown as CommonMo).noIC = true;
+      }
+      //
+      // Get information about the base element
+      //
+      this.baseScale = this.getBaseScale();
+      this.baseIc = this.getBaseIc();
+      this.baseIsChar = this.isCharBase();
+    }
+
+    /***************************************************************************/
+    /*
+     *  Methods for information about the core element for the base
+     */
+
+    /**
+     * @return {W}    The wrapper for the base core mi or mo (or whatever)
+     */
+    public getBaseCore(): W {
+      let core = this.getSemanticBase() || this.childNodes[0];
+      while (core && (core.childNodes.length === 1 &&
+              (core.node.isKind('mrow') || core.node.isKind('TeXAtom') ||
+               core.node.isKind('mstyle') || core.node.isKind('mpadded') ||
+               core.node.isKind('mphantom') || core.node.isKind('semantics')))) {
+        core = core.childNodes[0];
       }
+      return core || this.childNodes[0];
     }
 
     /**
-     * This gives the common bbox for msub and msup.  It is overridden
-     * for all the others (msubsup, munder, mover, munderover).
-     *
-     * @override
+     * @return {W}    The base fence item or null
      */
-    public computeBBox(bbox: BBox, recompute: boolean = false) {
-      const basebox = this.baseChild.getBBox();
-      const scriptbox = this.script.getBBox();
-      const [x, y] = this.getOffset(basebox, scriptbox);
-      bbox.append(basebox);
-      bbox.combine(scriptbox, bbox.w + x, y);
-      bbox.w += this.font.params.scriptspace;
-      bbox.clean();
-      this.setChildPWidths(recompute);
+    public getSemanticBase(): W {
+      let fence = this.node.attributes.getExplicit('data-semantic-fencepointer') as string;
+      return this.getBaseFence(this.baseChild, fence);
     }
 
     /**
-     * @return {number}  The ic for the core element
+     * Recursively retrieves an element for a given fencepointer.
+     *
+     * @param {W} fence The potential fence.
+     * @param {string} id The fencepointer id.
+     * @return {W} The original fence the scripts belong to.
      */
-    public coreIC(): number {
-      const corebox = this.baseCore.getBBox();
-      return (corebox.ic ? 1.05 * corebox.ic + .05 : 0);
+    public getBaseFence(fence: W, id: string): W {
+      if (!fence || !fence.node.attributes || !id) {
+        return null;
```

Comment:
Should that not simply return `fence`? This is related to my question above.

Reply:
Actually in this view it is below.

Reply:
No.  The three cases here are cases where the fence has not been found, but only the first one has `fence` being `null`, so if you returned `fence` in the other two cases, that would indicate that you found a semantic fence when you really didn't.  You only want to return the `fence` when it exists, is a node with attributes (e.g., not a text node), and the `data-semantic-id` attribute matches the (non-empty) `id` that you are looking for.  That is what line 322 below does.  

Note that this was [your code](https://github.com/mathjax/MathJax-src/pull/605/files#diff-827f9299f8e5cea1db7b2c0162aebec7f45941a24f73e6254d532b88156e0678) from your original PR (with the addition of the check for the attributes).